sysadmin/build.yml

```diff
@@ -0,0 +1,617 @@
+        group: solr
+        mode: 0770
+      when: not install_solr_data
+
+    - name: copy over schema.xml
+      copy:
+        src: "../solr-config/Solr-7/schema.xml"
+        dest: /var/solr/data/open-context/conf/schema.xml
+        owner: solr
+        group: solr
+        mode: 0770
+      when: not install_solr_data
+
+    # download the solr backup 
+    - name: download solr backup 
+      get_url:
+        url: "{{GROUP_SECRETS.SOLR_BACKUP_URL}}"
+        dest: "/tmp/{{GROUP_SECRETS.SOLR_BACKUP}}"
+        mode: 0640
+        owner: solr
+        group: solr
+      when: install_solr_data
+
+    - name: create /tmp/solr_backup
+      file: path="/tmp/solr_backup" state=directory owner=solr group=solr mode=0770
+      become: yes
+      when: install_solr_data
+
+    # unarchive the solr zip file
+    - name: unzip solr zip file
+      unarchive:
+        src: "/tmp/{{GROUP_SECRETS.SOLR_BACKUP}}"
+        dest: /tmp/solr_backup
+        remote_src: true
+        owner: solr
+        group: solr
+      when: install_solr_data
+
+    # # this didn't work -- time out on some privilege escalation
+    # - name: create open-context core 
+    #   command: "/opt/solr/bin/solr create_core -c open-context"
+    #   become: true
+    #   become_method: su
+    #   become_user: solr
+    #   become_flags: '-s /bin/bash'    
+
+    # mv /tmp/solr_backup/oc-solr-6/server/solr/open-context/* /var/solr/data/open-context/
+
+    - name: stat solr backup data 
+      stat: path=/tmp/solr_backup/oc-solr-6/server/solr/open-context
+      register: oc_stat
+      become: yes
+      when: install_solr_data
+
+    # TO DO -- parameterize oc-solr-6
+    - name: Move solr backup to solr data directory
+      command: sudo -u solr bash -c "rm -rf /var/solr/data/open-context/; mv -f /tmp/solr_backup/oc-solr-6/server/solr/open-context /var/solr/data/"
+      # when: oc_stat.exists and install_solr_data
+      when: install_solr_data
+
+    # issue command to reload core
+    - name: issue command to reload core
+      command: curl "http://localhost:8983/solr/admin/cores?action=RELOAD&core=open-context&&wt=json"
+      when: install_solr_data
+
+    # then need a reload https://serverfault.com/questions/703031/how-do-i-add-a-solr-core-without-restarting-the-solr-server
+    # curl "http://localhost:8983/solr/admin/cores?action=RELOAD&core=open-context&&wt=json"
+    # curl "http://octest.raymondyee.net:8983/solr/admin/cores?action=RELOAD&core=collection1&wt=json"
+
+    # - name: end play
+    #   meta: end_play
+ 
+    # - name: grab ssh config ip
+    #   debug:
+    #      var: "hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]"
+
+    # - name: Display all variables/facts known for a host
+    #   debug:
+    #     var: hostvars[inventory_hostname]
+    #   tags: debug_info
+
+    # - name: Getting PYTHONPATH
+    #   local_action: shell python -c 'import sys; print(":".join(sys.path))'
+    #   register: pythonpath
+
```

Comment:
for `opencontext`, there are still steps left to build an instance with test data, which then get fed into a solr index.  I might want to ansibilize those steps:


```
vagrant up opencontext --provider=google
```

After the machine comes up:

```
scp $(vagrant ssh-config opencontext |  tail -n +2 | awk '{if ($1) print " -o "$1"="$2;}') ~/Downloads/feb-2019-oc-test.backup localhost:/tmp
```

*Alternatively* -- login to machine and

```
wget -P /tmp https://dl.dropboxusercontent.com/s/0gnydnhw4dpco6c/feb-2019-oc-test.backup
```

login to machine 

```
vagrant ssh opencontext
cd /var/oc-venv
pipenv shell
```

Then:

```
python manage.py flush --noinput
```

Load the data:

```
PGPASSWORD=opencontextpw pg_restore -c -v -d opencontextdb -U opencontextuser -h localhost  -Fc -j 8 /tmp/feb-2019-oc-test.backup
```

then to set up solr:

```
sudo -u solr bash -c "/opt/solr/bin/solr delete -c open-context"
sudo -u solr bash -c "/opt/solr/bin/solr create_core -c open-context"

sudo cp /var/oc-venv/solr-config/Solr-7/schema.xml /var/solr/data/open-context/conf/schema.xml
#sudo -u solr bash -c "cp /var/oc-venv/solr-config/Solr-7/solrconfig_201808.xml /var/solr/data/open-context/conf/solrconfig.xml"
sudo cp /var/oc-venv/solr-config/Solr-7/solrconfig.xml /var/solr/data/open-context/conf/solrconfig.xml
sudo cp /var/oc-venv/solr-config/Solr-7/currency.xml /var/solr/data/open-context/conf/currency.xml
sudo cp /var/oc-venv/solr-config/Solr-7/elevate.xml /var/solr/data/open-context/conf/elevate.xml
sudo cp /var/oc-venv/solr-config/Solr-7/email_url_types.txt /var/solr/data/open-context/conf/email_url_types.txt

sudo chown solr:solr /var/solr/data/open-context/conf/*

curl "http://localhost:8983/solr/admin/cores?action=RELOAD&core=open-context&&wt=json"
sudo -u solr bash -c "/opt/solr/bin/solr restart"
```

and then fire up the django shell (` python manage.py shell`)


```Python
from opencontext_py.apps.ocitems.manifest.models import Manifest
from opencontext_py.apps.indexer.reindex import SolrReIndex
uuids = [m.uuid for m in Manifest.objects.all()]
print('Items to index:{} '.format(len(uuids)))
sri = SolrReIndex()
sri.reindex_uuids(uuids)
```